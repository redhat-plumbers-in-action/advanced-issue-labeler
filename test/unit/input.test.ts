import { describe, it, expect, beforeEach, test } from 'vitest';

import {
  inputsContextFixture,
  IInputsTestContext,
} from './fixtures/input.fixture';

describe('Inputs Object', () => {
  beforeEach<IInputsTestContext>(context => {
    context.inputs = inputsContextFixture.inputs;
  });

  it<IInputsTestContext>('can be instantiated', context =>
    context.inputs.map(inputsItem => expect(inputsItem).toBeDefined()));

  test<IInputsTestContext>('get template()', context =>
    context.inputs.map(inputsItem =>
      expect(inputsItem.template).toMatchSnapshot()
    ));

  test.todo('set template()');

  test<IInputsTestContext>('get section()', context =>
    context.inputs.map(inputsItem =>
      expect(inputsItem.section).toMatchSnapshot()
    ));

  test.todo('set section()');

  test<IInputsTestContext>('get blockList()', context =>
    context.inputs.map(inputsItem =>
      expect(inputsItem.blockList).toMatchSnapshot()
    ));

  test.todo('set blockList()');
});
