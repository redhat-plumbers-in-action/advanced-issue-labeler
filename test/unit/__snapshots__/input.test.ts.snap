// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Inputs Object > get blockList() 1`] = `[]`;

exports[`Inputs Object > get blockList() 2`] = `[]`;

exports[`Inputs Object > get blockList() 3`] = `
[
  "block1",
  "block2",
]
`;

exports[`Inputs Object > get blockList() 4`] = `[]`;

exports[`Inputs Object > get section() 1`] = `"section"`;

exports[`Inputs Object > get section() 2`] = `"section"`;

exports[`Inputs Object > get section() 3`] = `"section"`;

exports[`Inputs Object > get section() 4`] = `"section"`;

exports[`Inputs Object > get template() 1`] = `undefined`;

exports[`Inputs Object > get template() 2`] = `undefined`;

exports[`Inputs Object > get template() 3`] = `undefined`;

exports[`Inputs Object > get template() 4`] = `"template"`;
